packages/office-ui-fabric-react/src/utilities/dateMath/DateMath.ts

```diff
@@ -198,4 +198,100 @@ function getDatePartHashValue(date: Date) {
   /* tslint:disable:no-bitwise */
   return date.getDate() + (date.getMonth() << 5) + (date.getFullYear() << 9);
   /* tslint:enable:no-bitwise */
-}
\ No newline at end of file
+}
+
+/**
+ * Returns the week numbers for each week in a month.
+ * Week numbers are 1 - 52 (53) in a year
+ */
+export function getWeekNumbers(weeks: any[], firstDayOfWeek: DayOfWeek, navigatedDate: Date) {
+  let selectedYear = navigatedDate.getFullYear();
+  let selectedMonth = navigatedDate.getMonth();
+  let previousWeekNumber = 0;
+
+  // If first day of week is not Sunday, use alternative calculations
+  if (firstDayOfWeek === 0) {
+    let firstDayOfMonth = new Date(selectedYear, selectedMonth, 1);
+    let getDayofYear = getDayOfYear(addDays(firstDayOfMonth, firstDayOfWeek + 1));
+    previousWeekNumber = Math.ceil(getDayofYear / DAYS_IN_WEEK) - 1;
+  } else {
+    previousWeekNumber = getPreviousWeekNumbers(firstDayOfWeek, navigatedDate);
```

Comment:
Why do you need previous week numbers in order to calculate the current week's numbers? This seems wasteful. Did you take a look at OwaDateTime.GetFirstDayWeekOfYear in JSMVVM? The logic there seems a lot more compact and optimized and does not require looking up previous months week numbers.